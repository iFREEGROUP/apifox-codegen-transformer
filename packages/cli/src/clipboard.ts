import { execa } from 'execa'
import * as os from 'node:os'
import * as process from 'node:process'

export async function read(): Promise<string> {
  switch (os.platform()) {
    case 'win32': {
      const subProcess = await execa('Get-Clipboard', {
        shell: 'powershell.exe',
      })
      return subProcess.stdout
    }
    case 'linux':
      if (process.env.XDG_SESSION_TYPE === 'wayland') {
        const subProcess = await execa('wl-paste')
        return subProcess.stdout
      } else {
        const subProcess = await execa('xclip', ['-o'])
        return subProcess.stdout
      }
    default:
      throw new Error('Reading from clipboard is unsupported on this platform.')
  }
}

export async function write(content: string): Promise<void> {
  switch (os.platform()) {
    case 'win32':
      await execa('Set-Clipboard', [content], { shell: 'powershell.exe' })
      break
    case 'linux':
      if (process.env.XDG_SESSION_TYPE === 'wayland') {
        await execa('wl-copy', [content], { stdio: 'ignore' })
      } else {
        await execa('xclip', { input: content })
      }
      break
    default:
      throw new Error('Writing to clipboard is unsupported on this platform.')
  }
}
