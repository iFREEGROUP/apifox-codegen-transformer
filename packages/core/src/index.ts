import * as recast from 'recast'
import * as parser from 'recast/parsers/typescript.js'

export function transform(source: string): string {
  const ast = recast.parse(source, { parser })

  recast.visit(ast, {
    visitComment(path) {
      path.prune()
      return false
    },
    visitTSIndexSignature(path) {
      path.prune()
      return false
    },
  })

  return recast.print(ast).code
}
