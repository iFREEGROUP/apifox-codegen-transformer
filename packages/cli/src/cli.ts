#!/usr/bin/env node

import { transform } from '@apifox-codegen-transformer/core'
import getStdin from 'get-stdin'

const stdin = await getStdin()
console.log(transform(stdin))
