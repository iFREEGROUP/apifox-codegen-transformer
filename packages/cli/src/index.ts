#!/usr/bin/env node

import * as process from 'node:process'
import { run } from './cli.js'

await run(process.argv.slice(2))
