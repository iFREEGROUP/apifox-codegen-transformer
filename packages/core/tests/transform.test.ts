import { expect, test } from 'vitest'
import { transform } from '../src'

test('basic transformation', () => {
  const output = transform(`
export type T1 = {
    /**
     * field1
     */
    field1: string;
    /**
     * field2
     */
    field2: number;
    [property: string]: any;
}

export type T2 = {
    /**
     * field3
     */
    field3: string;
    /**
     * field4
     */
    field4: number;
    [key: string]: any;
}
`)
  expect(output).toMatchSnapshot()
})
