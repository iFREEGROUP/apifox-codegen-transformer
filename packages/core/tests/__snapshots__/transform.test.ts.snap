// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`basic transformation 1`] = `
"
export type T1 = {
 field1: string,
 field2: number
}

export type T2 = {
 field3: string,
 field4: number
}
"
`;
